step-api/step-api-reporting/src/main/java/step/reporting/LiveReporting.java

```diff
@@ -1,12 +1,41 @@
 package step.reporting;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import step.streaming.client.upload.StreamingUploadProvider;
 import step.streaming.client.upload.StreamingUploads;
+import step.streaming.client.upload.impl.local.DiscardingStreamingUploadProvider;
+import step.streaming.client.upload.impl.local.LocalDirectoryBackedStreamingUploadProvider;
+
+import java.io.File;
 
 public class LiveReporting {
+    public static class ConfigPropertyKeys {
+        // The CLI uses the same name for its configuration, this is why it's kept simple
+        public static final String STREAMING_UPLOADS_DIRECTORY = "streamingUploadsDir";
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(LiveReporting.class);
 
     public final StreamingUploads fileUploads;
 
-    public LiveReporting(StreamingUploads fileUploads) {
-        this.fileUploads = fileUploads; // attention: could be null in certain circumstances
+    public LiveReporting(StreamingUploadProvider streamingUploadProvider) {
+        if (streamingUploadProvider == null) {
+            logger.debug("LiveReporting initializing without a provided StreamingUploads object, locally instantiating one");
+            streamingUploadProvider = instantiateStreamingUploadProvider();
+        }
+        fileUploads = new StreamingUploads(streamingUploadProvider);
+    }
+
+    private StreamingUploadProvider instantiateStreamingUploadProvider() {
+        String directoryProperty = System.getProperty(ConfigPropertyKeys.STREAMING_UPLOADS_DIRECTORY, null);
+        if (directoryProperty == null) {
```

Comment:
By default, we shouldn't discard the data but use a temporary directory which would be deleted when the ExecutionContext of the KeywordRunner is closed. We might also add a method in the ExecutionContext to access the content of this directory. This would allow to read the produced files.

Doing this would also make the configurable directory unnecessary.